tools/gen_fury_builder_doc.py

```diff
@@ -0,0 +1,224 @@
+            else "None"
+        )
+        fval = (
+            field.field_default_val
+            if field.field_default_val is not None and len(field.field_default_val) > 0
+            else "None"
+        )
+
+        if len(fname) > hdr1_width:
+            hdr1_width = len(fname) + margin_right
+
+        if len(fdesc) > hdr2_width:
+            hdr2_width = len(fdesc) + margin_right
+
+        if len(fval) > hdr3_width:
+            hdr3_width = len(fval) + margin_right
+
+    table_header = (
+        f"|{hdr1}{(hdr1_width - len(hdr1)) * ' '}"
+        f"|{hdr2}{(hdr2_width - len(hdr2)) * ' '}"
+        f"|{hdr3}{(hdr3_width - len(hdr3)) * ' '}|"
+        f"\n"
+    )
+    table_delimiter = f"|{hdr1_width * '-'}|{hdr2_width * '-'}|{hdr3_width * '-'}|\n"
+    table_body = ""
+    for field in fields:
+        fname = field.field_name
+        fdesc = (
+            field.field_comment
+            if field.field_comment is not None and len(field.field_comment) > 0
+            else "None"
+        )
+        fval = (
+            field.field_default_val
+            if field.field_default_val is not None and len(field.field_default_val) > 0
+            else "None"
+        )
+
+        row = (
+            f"|`{fname}`{(hdr1_width - len(fname) - 2) * ' '}"
+            f"| {fdesc}{(hdr2_width - len(fdesc) - 1) * ' '}"
+            f"| {fval}{(hdr3_width - len(fval) - 1) * ' '}|"
+            f"\n"
+        )
+        table_body += row
+
+    table = table_header + table_delimiter + table_body
+    repl = f"{DOC_GEN_BEGIN}\n{table}\n{DOC_GEN_END}"
+    to_write = FILE_REPLACE_PATTERN.sub(repl, content)
+
+    with open(JAVA_DOC_PATH, "w") as f:
+        f.write(to_write)
+
+
+def main():
+    # 1. Try installing lxml
+    try_count = 3
+    while try_count >= 0:
+        try:
+            from lxml import etree
+        except Exception:
+            if try_count == 0:
+                return 1
+            print("Try installing lxml.")
+            subprocess.check_call("pip3 install lxml", shell=True)
+        finally:
+            try_count -= 1
+
+    # 2. Generating javadoc
+    print("Generating javadoc...")
+    tmp_dir = tempfile.gettempdir()
+    output_dir = "output"
+    subprocess.call(
+        f"cd {PROJECT_ROOT_DIR}java/fury-core;"
+        f"mvn javadoc:javadoc -DreportOutputDirectory={tmp_dir} -DdestDir={output_dir} -Dshow=private",
+        shell=True,
+        stdout=subprocess.DEVNULL,
+    )
+    print("javadoc generated successfully.")
+
+    # 3. Parsing javadoc
+    javadoc_dir = os.path.join(tmp_dir, output_dir)
+    fury_build_src = os.path.join(
+        javadoc_dir, "org/apache/fury/config/FuryBuilder.html"
+    )
+    with open(fury_build_src) as f:
+        content = f.read()
+    shutil.rmtree(javadoc_dir)
+
+    html = etree.HTML(content)
+    # There is only one `Field Detail`
+    field_detail = html.xpath('//div[@class="details"]//section[1]')[0]
+    if field_detail is None:
+        return 1
```

Comment:
how about raise exception with detailed message instead of exit by error code

Reply:
Update